modules/ethereum/src/import.rs

```diff
@@ -166,214 +166,221 @@ pub fn header_import_requires_receipts<S: Storage>(
 #[cfg(test)]
 mod tests {
 	use super::*;
-	use crate::tests::{
-		block_i, custom_block_i, genesis, signed_header, validator, validators_addresses, InMemoryStorage,
+	use crate::mock::{
+		block_i, custom_block_i, custom_test_ext, genesis, signed_header, test_aura_config, test_validators_config,
+		validator, validators, validators_addresses, TestRuntime,
 	};
 	use crate::validators::ValidatorsSource;
-	use crate::{kovan_aura_config, kovan_validators_config};
+	use crate::{BridgeStorage, Headers, OldestUnprunedBlock};
+	use frame_support::{StorageMap, StorageValue};
 
 	#[test]
 	fn rejects_finalized_block_competitors() {
-		let mut storage = InMemoryStorage::new(genesis(), validators_addresses(3));
-		storage.finalize_headers(Some((100, Default::default())), None);
-		assert_eq!(
-			import_header(
-				&mut storage,
-				&kovan_aura_config(),
-				&kovan_validators_config(),
-				PRUNE_DEPTH,
-				None,
-				Default::default(),
-				None,
-			),
-			Err(Error::AncientHeader),
-		);
+		custom_test_ext(genesis(), validators_addresses(3)).execute_with(|| {
+			let mut storage = BridgeStorage::<TestRuntime>::new();
+			storage.finalize_headers(Some((100, Default::default())), None);
+			assert_eq!(
+				import_header(
+					&mut storage,
+					&test_aura_config(),
+					&test_validators_config(),
+					PRUNE_DEPTH,
+					None,
+					Default::default(),
+					None,
+				),
+				Err(Error::AncientHeader),
+			);
+		});
 	}
 
 	#[test]
 	fn rejects_known_header() {
-		let validators = (0..3).map(|i| validator(i as u8)).collect::<Vec<_>>();
-		let mut storage = InMemoryStorage::new(genesis(), validators_addresses(3));
-		let block = block_i(&storage, 1, &validators);
-		assert_eq!(
-			import_header(
-				&mut storage,
-				&kovan_aura_config(),
-				&kovan_validators_config(),
-				PRUNE_DEPTH,
-				None,
-				block.clone(),
-				None,
-			)
-			.map(|_| ()),
-			Ok(()),
-		);
-		assert_eq!(
-			import_header(
-				&mut storage,
-				&kovan_aura_config(),
-				&kovan_validators_config(),
-				PRUNE_DEPTH,
-				None,
-				block,
-				None,
-			)
-			.map(|_| ()),
-			Err(Error::KnownHeader),
-		);
+		custom_test_ext(genesis(), validators_addresses(3)).execute_with(|| {
+			let validators = validators(3);
+			let mut storage = BridgeStorage::<TestRuntime>::new();
+			let block = block_i(1, &validators);
+			assert_eq!(
+				import_header(
+					&mut storage,
+					&test_aura_config(),
+					&test_validators_config(),
+					PRUNE_DEPTH,
+					None,
+					block.clone(),
+					None,
+				)
+				.map(|_| ()),
+				Ok(()),
+			);
+			assert_eq!(
+				import_header(
+					&mut storage,
+					&test_aura_config(),
+					&test_validators_config(),
+					PRUNE_DEPTH,
+					None,
+					block,
+					None,
+				)
+				.map(|_| ()),
+				Err(Error::KnownHeader),
+			);
+		});
 	}
 
 	#[test]
 	fn import_header_works() {
```

Comment:
I think this test is either doing too much, or the name needs to be updated.

If we're just checking that importing headers work, I don't think we need that `Multi` authority set thing that's going on. That could be part of a different test that checks validator set changes.

Reply:
Logged #111